docs/ds/dividing.md

```diff
@@ -48,16 +48,16 @@ begin
     end;
     toleft[deep,i]:=toleft[deep,i-1]+flag;
   end;
-  Build(left,mid,deep+1); // 继续
+  Build(left,mid,deep+1); // continue
   Build(mid+1,right,deep+1);
 end;
 ```
 
-## 查询
+## Querying
 
-那我们先扯一下主席树的内容。在用主席树求区间第 $K$ 小的时候，我们以 $K$ 为基准，向左就向左，向右要减去向左的值，在划分树中也是这样子的。
+Let's first talk about the content of the persistent segment tree. When using the persistent segment tree to find the smallest $K$ in the interval, we use $K$ as the benchmark. If going to left, we go to left; if going to right, we must to subtract the value of the left. This is also the case in the dividing tree.
 
-查询难理解的，在于 **区间缩小** 这种东西。下图，我查询的是 $3$ 到 $7$ , 那么下一层我就只需要查询 $2$ 到 $3$ 了。当然，我们定义 $left,right$ 为缩小后的区间（目标区间）， $l,r$ 还是我所在节点的区间。那为什么要标出目标区间呢？因为那是 **判定答案在左边，右边的基准** 。
+What makes the query difficult to understand is the **interval reduction**. In the figure below, we are querying from $3$ to $7$ , so we only need to query from $2$ to $3$ at the next level. Of course, we define $left,right$ as the reduced interval (target interval), and $l,r$ as the interval of the node. Then why should we mark the target interval? Because that's **the benchmark to determine whether the answer is on the left or on the right**.
```

Comment:
```suggestion
What makes the query difficult to understand is the **interval reduction**. In the figure below, we are querying from $3$ to $7$ , so we only need to query from $2$ to $3$ at the next layer. Of course, we define left and right as the reduced interval (target interval), and $l,r$ as the interval of the node. Then why should we mark the target interval? Because that's **the benchmark to determine whether the answer is on the left or on the right**.
```